src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -307,65 +321,79 @@ protected function guardAgainstInvalidTranslationResetRequest($content)
     /**
      * @param $lang
      * @param $type
-     * @param $selected
+     * @param string $theme Selected theme name
      * @param null $search
      *
      * @return array
      */
-    private function getNormalTree($lang, $type, $selected, $search = null)
+    private function getNormalTree($lang, $type, $theme, $search = null)
     {
-        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $selected);
-        $catalogue = $this->translationService->getTranslationsCatalogue($lang, $type, $selected, $search);
+        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $theme);
+        $catalogue = $this->translationService->getTranslationsCatalogue($lang, $type, $theme, $search);
 
-        return $this->getCleanTree($treeBuilder, $catalogue, $type, $selected, $search);
+        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search);
     }
 
     /**
-     * @param $lang
-     * @param $type
-     * @param $selected
-     * @param null $search
+     * @param string $lang Two-letter iso code
+     * @param string $selectedModuleName Selected module name
+     * @param string|null $search
      *
      * @return array
      */
-    private function getModulesTree($lang, $type, $selected, $search = null)
+    private function getModulesTree($lang, $selectedModuleName, $search = null)
```

Comment:
I suggest same defensive logic: if `$selectedModuleName` is illogical, we crash. Translations are a mess right now, this would clear the path and avoid ambiguous successfull behaviors

Reply:
(maybe not in a patch version though 😛)

Reply:
I think it will crash but deeper in the call stack. But we agree that this whole thing needs to be refactored.